### ISSUE TITLE
Require upper bounds message in 2961.x update of plugin bom

### ISSUE THREAD

**Issue by MarkEWaite** [jenkinsci/bom#3098]
### Jenkins and plugins versions report

<details>
  <summary>Ubuntu 22.04 Development Environment</summary>

  
  ```text
Specific plugins are not relevant
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Ubuntu 22.04

### Reproduction steps

1. Updating the plugin bom version to [2961.v1f472390972e](https://github.com/jenkinsci/bom/releases/tag/2961.v1f472390972e) in several plugins fails with an upper bounds message if the plugins include scm-api or json-path-api as one of their transient dependencies and if they require Jenkins 2.426.3 or newer as their minimum Jenkins version.  The plugins that I've seen show the message include:
    * https://github.com/jenkinsci/basic-branch-build-strategies-plugin/pull/77 
    * https://github.com/jenkinsci/elastic-axis-plugin/pull/362
    * https://github.com/jenkinsci/nodelabelparameter-plugin/pull/302
    * https://github.com/jenkinsci/testng-plugin-plugin/pull/281

### Expected Results

No upper bounds message reported

### Actual Results

For the elastic axis plugin job on [ci.jenkins.io](https://ci.jenkins.io/job/Plugins/job/elastic-axis-plugin/job/PR-362/) , the report is:
```
Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.4.1:enforce (display-info) on project elastic-axis:
Rule 5: org.apache.maven.enforcer.rules.dependency.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.ow2.asm:asm:9.6 paths to dependency are:
+-org.jenkins-ci.plugins:elastic-axis:563.va_deb_9db_70f06
  +-org.jenkins-ci.plugins:matrix-project:822.824.v14451b_c0fd42
    +-org.jenkins-ci.plugins:junit:1265.v65b_14fa_f12f0 (managed) <-- org.jenkins-ci.plugins:junit:1240.vf9529b_881428
      +-org.jenkins-ci.plugins.workflow:workflow-api:1291.v51fd2a_625da_7 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-api:1283.v99c10937efcb_
        +-org.jenkins-ci.plugins:scm-api:689.v237b_6d3a_ef7f (managed) <-- org.jenkins-ci.plugins:scm-api:672.v64378a_b_20c60
          +-io.jenkins.plugins:asm-api:9.7-33.v4d23ef79fcc8 (managed) <-- io.jenkins.plugins:asm-api:9.6-3.v2e1fa_b_338cd7
            +-org.ow2.asm:asm:9.6 (managed) <-- org.ow2.asm:asm:9.7

```

The pull request is:

* https://github.com/jenkinsci/elastic-axis-plugin/pull/362

### Anything else?

The git plugin and the git client plugin did not report the upper bounds message because they currently only require Jenkins 2.414.3 as their minimum Jenkins version.  When I update a local copy to require Jenkins 2.426.x or newer, then git plugin and git client plugin report the same upper bounds error message  

When I update a local copy to require Jenkins 2.452 or newer and to use bom-weekly, the upper bounds message does not appear.  I believe that is because Jenkins 2.452 includes asm 9.7.

### Are you interested in contributing a fix?

Yes, although I am not confident of the preferred way to resolve the issue.

If I add an exclusion of `asm-api` to the `scm-api` dependency in the pom file of the plugin, it compiles successfully but fails tests because the scm-api plugin cannot be loaded without the `asm-api` plugin.

If I add an explicit dependency on `asm-api` plugin, it does not seem to resolve the issue.

**Comment by viceice** [comment]
same here
- https://github.com/jenkinsci/scoring-load-balancer-plugin/pull/32

**Comment by basil** [comment]
https://github.com/jenkinsci/plugin-pom/pull/916, released in 4.81, contains a temporary workaround. The long-term fix will be to remove ASM from Jenkins core.
